lib/Runtime/Library/JsBuiltInEngineInterfaceExtensionObject.cpp

```diff
@@ -0,0 +1,410 @@
+            if (scriptContext->IsProfiling())
+            {
+                scriptContext->RegisterScript(functionGlobal->GetFunctionProxy());
+            }
+#endif
+
+#ifdef ENABLE_SCRIPT_DEBUGGING
+            // Mark we are profiling library code already, so that any initialization library code called here won't be reported to profiler.
+            // Also tell the debugger not to record events during intialization so that we don't leak information about initialization.
+            AutoInitLibraryCodeScope autoInitLibraryCodeScope(scriptContext);
+#endif
+
+            Js::Var args[] = { scriptContext->GetLibrary()->GetUndefined(), scriptContext->GetLibrary()->GetEngineInterfaceObject() };
+            Js::CallInfo callInfo(Js::CallFlags_Value, _countof(args));
+
+            // Clear disable implicit call bit as initialization code doesn't have any side effect
+            Js::ImplicitCallFlags saveImplicitCallFlags = scriptContext->GetThreadContext()->GetImplicitCallFlags();
+            scriptContext->GetThreadContext()->ClearDisableImplicitFlags();
+            JavascriptFunction::CallRootFunctionInScript(functionGlobal, Js::Arguments(callInfo, args));
+            scriptContext->GetThreadContext()->SetImplicitCallFlags((Js::ImplicitCallFlags)(saveImplicitCallFlags));
+
+            Js::ScriptFunction *functionBuiltins = scriptContext->GetLibrary()->CreateScriptFunction(jsBuiltInByteCode->GetNestedFunctionForExecution(0));
+            functionBuiltins->SetPrototype(scriptContext->GetLibrary()->nullValue);
+
+            // Clear disable implicit call bit as initialization code doesn't have any side effect
+            saveImplicitCallFlags = scriptContext->GetThreadContext()->GetImplicitCallFlags();
+            scriptContext->GetThreadContext()->ClearDisableImplicitFlags();
+            JavascriptFunction::CallRootFunctionInScript(functionBuiltins, Js::Arguments(callInfo, args));
+            scriptContext->GetThreadContext()->SetImplicitCallFlags((Js::ImplicitCallFlags)(saveImplicitCallFlags));
+
+            InitializePrototypes(scriptContext);
+#if DBG_DUMP
+            if (PHASE_DUMP(Js::ByteCodePhase, functionGlobal->GetFunctionProxy()) && Js::Configuration::Global.flags.Verbose)
+            {
+                DumpByteCode();
+            }
+#endif
+        }
+        catch (const JavascriptException& err)
+        {
+            Js::JavascriptExceptionObject* ex = err.GetAndClear();
+            ex->GetScriptContext();
+            JavascriptError::ThrowTypeError(ex->GetScriptContext(), JSERR_BuiltInNotAvailable);
+        }
+    }
+
+    bool JsBuiltInEngineInterfaceExtensionObject::InitializeJsBuiltInNativeInterfaces(DynamicObject * builtInNativeInterfaces, DeferredTypeHandlerBase * typeHandler, DeferredInitializeMode mode)
+    {
+        typeHandler->Convert(builtInNativeInterfaces, mode, 16);
```

Comment:
>16 [](start = 60, length = 2)

curious why 16?

Reply:
Its the initSlotCapacity for the SimpleDictionaryTypeHandler. Changed it to 4, I think that suffices.